cmake/CPM_boost.cmake

```diff
@@ -0,0 +1,35 @@
+
+set(ENV{CPM_SOURCE_CACHE} "${CMAKE_SOURCE_DIR}/tt_metal/third_party/boost")
```

Comment:
is this to prevent re-downloading of `boost` src whenever someone runs `CMake`? If this wasn't set, where does it default to?

Reply:
Yes, that is what sets the caching and download path. My intention is to not let devs set it so it will always download to `third_party/boost`

Reply:
where does it download to by default? build folder? How much time does this add? I prefer not to add caching (if possible) as it may introduce cache staleness issues

Reply:
By default it goes to build folder. It'll add ~10-20 seconds every time you configure a new cmake project (whenever someone runs `cmake -B build`) or anytime something causes the cmake to reconfigure. 

The caching can also be something that devs pass in themselves if that's what's preferred.

Reply:
Let's say you bump boost version after cache has been populated, is CPM smart enough to re-download newer version?

Reply:
Yes, it will.

Reply:
OK, I'm thinking we store it in `.cpmcache` or something similar in the root directory

Reply:
Ok switched to store in `.cpmcache`